### DISCUSSION TITLE
Is metering feature supported on WASI?

### DISCUSSION THREAD

**Discussion by jmb-51** [wasmerio/wasmer#2671] [Q&A]
Hi All,

I have tried to combine examples of metering and wasi into a test crate to investigate if it is possible to do so.
[Metering example from wasmer](https://github.com/wasmerio/wasmer/blob/master/examples/metering.rs)
[WASI example from wasmer](https://github.com/wasmerio/wasmer/blob/master/examples/wasi.rs)

I have also used the sample hello wasi wasm code found [here](https://github.com/wasmerio/wasmer/blob/master/tests/wasi-wast/wasi/snapshot1/hello.wasm).

However when I added the metering middleware into the wasi example, I am unable to call the function defined in the wasm code. 
The metering middleware is pushed by Cranelift compiler and I am using Universal engine.

TLDR:
I would appreciate if there are pointers/hints on how to debug the stack trace output from wasmer.
I would like to know if metering is supported in Wasmer's WASI implementation?

Thanks a lot for the help.

Files:
[cranelift-wasi-metering.tar.gz](https://github.com/wasmerio/wasmer/files/7502105/cranelift-wasi-metering.tar.gz)
[result.log](https://github.com/wasmerio/wasmer/files/7502107/result.log)

**Comment by jcaesar** [comment]
Metering triggers an `unreachable` instruction when it runs out of points. Since you've only given it 10 points, I suspect that's what's happening. Try starting with 10¹⁰ points, that should be good for at least a second of execution.
